server/src/modules/intentions/queries/findDemandes.query.ts

```diff
@@ -1,11 +1,35 @@
+import { sql } from "kysely";
+
 import { kdb } from "../../../db/db";
 import { cleanNull } from "../../../utils/noNull";
 
-export const findDemandes = async () => {
+export const findDemandes = async ({
+  status,
+  offset = 0,
+  limit = 20,
+  orderBy,
+}: {
+  status?: "draft" | "submitted";
+  offset?: number;
+  limit: number;
+  orderBy?: { order: "asc" | "desc"; column: string };
+}) => {
   const demandes = await kdb
     .selectFrom("demande")
     .selectAll()
```

Comment:
```suggestion
    .selectAll()
    .select(sql<string>`COUNT(*) OVER()`.as("count"))
```
Et l'accès à la valeur se fait comme ça:
```typescript
demandes[0]?.count ?? 0
```
Et en number:
```typescript
parseInt(demandes[0]?.count) || 0
```